pykafka/balancedconsumer.py

```diff
@@ -202,7 +201,8 @@ def checker():
                 time.sleep(120)
                 if not self._running:
                     break
-                self._check_held_partitions()
+                if not self._check_held_partitions():
+                    self._rebalance()
```

Comment:
Hah, cool, wasn't intended as an improvement, just that I'm lazy so I wanted to reuse that method for the test :)